openmc/mesh.py

```diff
@@ -961,6 +961,44 @@ def build_cells(self, bc=None):
 
         return root_cell, cells
 
+
+    def get_index_where(self, value: float, basis: str = 'xy'):
+        """Gets the mesh cell index nearest to the specified axis value.
+
+        Parameters
+        ----------
+        basis : {'xy', 'xz', 'yz'}
+            The basis directions for the slice
+        value : str
+            A string for the type of value to return  - 'mean' (default),
+            'std_dev', 'rel_err', 'sum', or 'sum_sq' are accepted
```

Comment:
I haven't been following this PR closely so apologies if I'm missing something. Just reading the description here, I'm really confused as to what this method is supposed to do or why a user would want to call it.